### DISCUSSION TITLE
Cuda initialization error when doing PyTorch inference with FastAPI

### DISCUSSION THREAD

**Discussion by eladrich** [fastapi/fastapi#8967] [Questions]
Hi,
I've been trying to use FastAPI to wrap a PyTorch model for inference.
While this works perfectly for CPU inference, when I'm trying to run my model on a GPU I encounter an `Cuda error: initialization error.` when trying to move a tensor from CPU to GPU.
I am able to load the model into the GPU during the `__init__` stage, so it looks like this problem is related to multiprocessing. However, standard solutions related to PyTorch such as setting its multiprocessing mode to `spawn` do not seem to work.

Any idea how to approach this from the `FastAPI` end?

**Comment by Mause** [comment]
Can you explain how your issue is related to fastapi?

**Comment by SaeedArisha** [comment]
@eladrich Did you manage to find a solution?

**Comment by shanmukh-gupta** [comment]
@SaeedArisha @eladrich Did you find a solution for this? I face the same error.
